ffyaml/ffyaml_test.go

```diff
@@ -71,3 +71,75 @@ func TestParser(t *testing.T) {
 		})
 	}
 }
+
+func TestParserAllTypesEmptyVals(t *testing.T) {
+	t.Parallel()
+
+	for _, testcase := range []struct {
+		name string
+		file string
+		want fftest.Vars
+	}{
+		{
+			name: "basic empty vals",
+			file: "testdata/empty_basic_vals.yaml",
+			want: fftest.Vars{S: "", I: 0, F: 0.00, D: 0 * time.Second, B: false, X: nil},
+		},
+	} {
+		t.Run(testcase.name, func(t *testing.T) {
+			fs, testVars := fftest.EmptyValsFS()
+			testVars.ParseError = ff.Parse(fs, []string{},
+				ff.WithConfigFile(testcase.file),
+				ff.WithConfigFileParser(ffyaml.Parser),
+			)
+			pError := testVars.ParseError
+			checkParseErr(t, pError)
+
+			if err := fftest.Compare(&testcase.want, testVars); err != nil {
+				t.Fatal(err)
+			}
+		})
+	}
+}
+
+func checkParseErr(t *testing.T, pError error) {
+	if pError != nil {
+		var errMsg string
+		errMsg = "ParseError: " + pError.Error()
+		if errors.Is(pError, ff.StringConversionError{}) {
+			errMsg = errMsg + ";errType: " + "StringConversionError"
+		}
+		t.Fatal(errMsg)
+	}
+}
+
+func TestEmptyValsDontOverwritePresets(t *testing.T) {
```

Comment:
Why does this need to be a separate test, rather than part of TestParser?

Reply:
> Why does this need to be a separate test, rather than part of TestParser?

This test function covers the "Negative-test scenario" that confirms that the No-Op for the Flagset update would not "overwrite" preset set of flags that are defined with something Other Than the Zero-Value for each type.

In order to do that, I needed to use a different test fixture method to set up the test, `PresetValsFS()`. This function sets the values of the Flagset variables to something other than the Zero value defaults for each type in golang. 

So on line 132, I call:

`fs, testVars := fftest.PresetValsFS()`

The test fixture, `PresetValsFS()`, in combination with test function `TestEmptyValsDontOverwritePresets()` ensure that the Negative-Test-Don't-Overwrite-Pre-defined-FlagSet-Values test case is covered.

Reply:
Test has been removed. see latest.